daemon/mgr/container.go

```diff
@@ -293,10 +296,36 @@ func (mgr *ContainerManager) Create(ctx context.Context, name string, config *ty
 		return nil, errors.Wrapf(errtypes.ErrInvalidParam, "NetworkingConfig cannot be empty")
 	}
 
-	id, err := mgr.generateID()
-	if err != nil {
-		return nil, err
+	if config.SpecificID != "" {
```

Comment:
I think the function `Create` function is so long, could you please put the ID generation part encapsulated into a single function? @wangforthinker

Reply:
Have Done it!